megamek/src/megamek/common/Jumpship.java

```diff
@@ -471,19 +628,135 @@ public double primitiveFuelFactor() {
     public String[] getLocationNames() {
         return LOCATION_NAMES;
     }
-
+    
+    //Methods for dealing with the K-F Drive, Sail and L-F Battery
+    
+    //Set the current KF Drive integrity
     public void setKFIntegrity(int kf) {
         kf_integrity = kf;
     }
-
+    
+    //Return the current KF Drive integrity
     public int getKFIntegrity() {
         return kf_integrity;
     }
+    
+    //Set the original/undamaged KF Drive integrity
+    public void setOKFIntegrity(int kf) {
+        original_kf_integrity = kf;
+    }
+    
+    //Return the original/undamaged KF Drive integrity
+    public int getOKFIntegrity() {
+        return original_kf_integrity;
+    }
+    
+    //Return the damage taken to the KF Drive
+    public int getKFDriveDamage() {
+        return (getOKFIntegrity() - getKFIntegrity());
+    }
+    
+    //Is any part of the KF Drive damaged?  Used by MHQ for repairs.
+    public boolean isKFDriveDamaged() {
+        return (getKFHeliumTankHit() 
+                || getKFDriveCoilHit() 
+                || getKFDriveControllerHit() 
+                || getLFBatteryHit() 
+                || getKFChargingSystemHit()
+                || getKFFieldInitiatorHit());
+    }
+    
+    //Set the 2/3 Original KF Drive integrity representing the helium tanks
+    public void setKFHeliumTankIntegrity(int ht) {
```

Comment:
Is the 2/3 comment meant to be for the caller? Or just understood.

If the caller must make that conversion, you should make this a javadoc.

If it would be easier to remember, you could do the math here instead.

Reply:
It was just supposed to be descriptive. I'm using some old BattleSpace rules to flesh out the KF Drive system more than SO does (same principle as Aero Sanity), so I wanted to give some insight into what's going on and why.